pkgs/tools/security/metasploit/default.nix

```diff
@@ -26,6 +26,10 @@ in stdenv.mkDerivation rec {
     sha256 = "1vilyy0dqzp8kbbpvs2zrv2ac7s39w2vv7mrbzgcjgh2bj7c6bg1";
   };
 
+  preBuild = ''
+    export HOME=$TEMPDIR
+  '';
```

Comment:
It actually needs to be applied in `gem-config` since it is affecting not metasploit but the `rb-readline` gem.

```diff
diff --git a/pkgs/development/ruby-modules/gem-config/default.nix b/pkgs/development/ruby-modules/gem-config/default.nix
index b5aa0933c2e..5d6cc04c44c 100644
--- a/pkgs/development/ruby-modules/gem-config/default.nix
+++ b/pkgs/development/ruby-modules/gem-config/default.nix
@@ -371,6 +371,9 @@ in
     postPatch = ''
       substituteInPlace lib/rbreadline.rb \
         --replace 'infocmp' '${ncurses.dev}/bin/infocmp'
+
+      # wants that at build time
+      export HOME=$TEMPDIR
     '';
   };
```diff

Reply:
With change I was able to build metasploit.

Reply:
If it's getting to /nix/store/qhinmjkf4zhyqpq4ffc2s2rqpbfn5wk9-ruby2.5.3-metasploit-framework-4.16.1.drv, shouldn't that mean it's already built rb-readline?

Reply:
I would say yes. I am not sure how this behaves on non-linux machines.
All I can say is that with the change above I can build it but not without.

Reply:
On Darwin your fix doesn't seem to solve the problem, but also neither does mine… I'll try pushing your fix, but then if you could get OfBorg to do a Darwin rebuild (I assume you have perms), that'd be great.